### PR TITLE
Nawqaan 441

### PR DESCRIPTION
**Pull request by mbucknell**
The non trivial changes were to filter the states list to those which contain data and changing the start/end times to use select2 to select a water year.

### CODE REVIEW

File: nar_common/static/nar_ui/download.js [hunk 1]
```diff
@@ -237,28 +260,79 @@ nar.downloads = (function() {
...
-		});
-		
-		$("#startDateTime").datepicker(datepickerOptions);
-		$("#endDateTime").datepicker(datepickerOptions);
+		var START_YEARS_LIST = [];
+		var END_YEARS_LIST = [];
+		var i;
+		for (i = yearLimits.minYear; i <= yearLimits.maxYear; i++) {
+			START_YEARS_LIST.push({
+				id : '10/01/' + (i - 1),
```
Comment (jiwalker-usgs) [inline nar_common/static/nar_ui/download.js]: Might want to move some of this functionality to WaterYearUtils to make it obvious where the id dates are coming from.  Also, ISO8601 would be preferred if possible.

Comment (jiwalker-usgs) [inline nar_common/static/nar_ui/download.js]: Though I don't think it is because of how dates are parsed on the service. :(

Comment (mbucknell) [inline nar_common/static/nar_ui/download.js]: Was just about to reply ... I used the same format as it was before.

**Comment by jiwalker-usgs** [comment]
Looks good, I should have probably changed the date format on the download service when I first noticed that, but we know what we mean. :shipit:

**Pull request merged**